apis/object/v1alpha1/types.go

```diff
@@ -81,6 +83,27 @@ type ObjectStatus struct {
 	AtProvider          ObjectObservation `json:"atProvider,omitempty"`
 }
 
+// ConnectionDetail points to a resource, from which a value should be copied.
+// This value will be saved in connection secret.
+type ConnectionDetail struct {
+	v1.ObjectReference    `json:",inline"`
+	DecodeBase64          bool   `json:"decodeBase64,omitempty"`
+	JSONPath              string `json:"JSONPath,omitempty"`
```

Comment:
@ppyt-pl For the reason of introducing `JSONPath`, IIRC, it's because when using `fieldPath` to specify where to get the value, it's hard to differentiate the key name from the path if the key name has dot(s). In this case, can we use something similar as below:

`data[.config.yml]` - The `.config.yml` field of the `data` object

This is documented [here](https://crossplane.io/docs/v1.5/reference/composition.html).

Reply:
@morningspace thank you for the review. I actually wanted to have a placeholder for implementation of `JSONPath` and its not implemented yet - for now let's forget about it and we can delete it from this struct (also `DecodeBase64` is not needed for now).
So the problem of retrieving key name with dot(s) appeared only in Secrets so I solved it by adding `fromSecretKey`. This consumes exactly the key name that is supplied and does not parsing the path. In opposite to `fieldPath` where it decodes the path based on dots.
I'll investigate it a bit more with the documentation you provided.

Reply:
@morningspace fixed. I reduced the types to match helm provider. Also now `fieldPath: data[tls.crt]` is handled correctly, thank you for suggestion!